### PR TITLE
Subsume welcome message

### PR DESCRIPTION
**Pull request by modalduality**

### CODE REVIEW

File: go/chat/helper.go [hunk 1]
```diff
@@ -18,168 +18,71 @@ import (
...
-	kuid, err := CurrentUID(s.G())
-	if err != nil {
-		return err
-	}
-	uid := gregor1.UID(kuid.ToBytes())
-
-	oneChatPerTLF := true
-	convs, _, err := FindConversations(ctx, s.G(), s.DebugLabeler, s.remoteInterface, uid, s.canonicalName,
-		chat1.TopicType_CHAT, s.membersType, chat1.TLFVisibility_PRIVATE, s.topicName, &oneChatPerTLF)
+func NewSendHelper(ctx context.Context, server *Server, nclArg chat1.NewConversationLocalArg) (*SendHelper, error) {
```
Comment (modalduality) [inline go/chat/helper.go]: rationale for taking in `NewConversationLocalArg` is that it's exactly the information we need, so there's no point in exploding it.

File: go/chat/helper.go [hunk 2]
```diff
@@ -18,168 +18,71 @@ import (
...
+		TopicName:        topicName,
+		IdentifyBehavior: identifyBehavior,
+		TopicType:        chat1.TopicType_CHAT,
+		TlfVisibility:    chat1.TLFVisibility_PRIVATE,
+		MembersType:      chat1.ConversationMembersType_TEAM,
+	}
 }
 
-type sendHelper struct {
+type SendHelper struct {
```
Comment (modalduality) [inline go/chat/helper.go]: Old architecture ran NewConversationLocal everytime a chat message was sent.
Here, it's only run once on initialization, which is why constructors can return errors.

File: go/chat/helper.go [hunk 3]
```diff
@@ -18,168 +18,71 @@ import (
...
-		TopicType: chat1.TopicType_CHAT,
-	}
-	var err error
-	s.triple.TopicID, err = utils.NewChatTopicID()
-	if err != nil {
-		return err
-	}
-
-	first := chat1.MessagePlaintext{
+func (s *SendHelper) NewPlaintextMessage(body string) chat1.MessagePlaintext {
```
Comment (modalduality) [inline go/chat/helper.go]: separated from sending for easy composition with other message types

File: go/chat/server.go [hunk 4]
```diff
@@ -491,6 +491,9 @@ func (h *Server) NewConversationLocal(ctx context.Context, arg chat1.NewConversa
 		IdentifyBehavior: arg.IdentifyBehavior,
 	}
 	if arg.TopicName != nil {
+		if arg.MembersType == chat1.ConversationMembersType_KBFS {
```
Comment (mmaxim) [inline go/chat/server.go]: Why is this here?

Comment (mmaxim) [inline go/chat/server.go]: This isn't actually true (you can for topicType == chat1.TopicType_DEV)

File: go/chat/helper.go [hunk 5]
```diff
@@ -18,168 +18,71 @@ import (
 	"github.com/keybase/client/go/protocol/keybase1"
 )
 
-func SendTextByName(ctx context.Context, g *globals.Context, name string, topicName string,
```
Comment (mmaxim) [inline go/chat/helper.go]: Can we keep these functions? Its nice to be able to just send a message without having to create an object.

File: go/chat/helper.go [hunk 6]
```diff
@@ -18,168 +18,71 @@ import (
...
-	boxer := NewBoxer(s.G())
-	sender := NewBlockingSender(s.G(), boxer, nil, s.remoteInterface)
-	msg := chat1.MessagePlaintext{
-		ClientHeader: chat1.MessageClientHeader{
-			Conv:        s.triple,
-			TlfName:     s.canonicalName,
-			MessageType: mtype,
+		MessageBody: chat1.MessageBody{
+			MessageType__: chat1.MessageType_TEXT,
+			Text__: &chat1.MessageText{
```
Comment (mmaxim) [inline go/chat/helper.go]: Don't create the body like this, use `chat1.NewMessageBodyText()`
